src/EditorFeatures/CSharpTest/Diagnostics/MakeMethodAsynchronous/MakeMethodAsynchronousTests.cs

```diff
@@ -34,15 +35,77 @@ public static void Test()
 @"using System;
 using System.Threading.Tasks;
 
-class Program 
+class Program
 {
-    public static async void TestAsync() 
+    public static async void TestAsync()
     {
         await Task.Delay(1);
     }
 }";
             await TestInRegularAndScriptAsync(initial, expected, index: 1);
         }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeMethodAsynchronous)]
+        [WorkItem(26312, "https://github.com/dotnet/roslyn/issues/26312")]
+        public async Task AwaitInVoidMainMethodWithModifiers()
+        {
+            var initial =
+@"using System;
+using System.Threading.Tasks;
+
+class Program
+{
+    public static void Main()
+    {
+        [|await Task.Delay(1);|]
+    }
+}";
+
+            var expected =
+@"using System;
+using System.Threading.Tasks;
+
+class Program
+{
+    public static async void Main()
+    {
```

Comment:
please add another test for the version of this feature that changes the return type to Task
![image](https://user-images.githubusercontent.com/11444821/39090178-141a70ae-45d8-11e8-89d7-fc43b303e26c.png)
by removing the "index: 1"

Reply:
also, i would personally consider disabling the "stay void" version entirely here because the resulting code is really bad